### ISSUE TITLE
Problem with Attention manipulation in the Code

### ISSUE THREAD

**Issue by aneesurhashmi** [shape-guided-diffusion/shape-guided-diffusion#4]
Hi, 
Thanks for the amazing work. 

I am trying to understand how you mask the cross and self-attention maps. Can you please share some details on how you access the CrossAttention when there is no module named "CrossAttention" (utils.py) in the "UNet2DConditionModel" class? (It uses UNetMidBlock2DCrossAttn, CrossAttnDownBlock2D, and CrossAttnUpBlock2D modules instead).  

One more thing, I tried commenting out all the places where you updated the attention maps e.g. save_mask_image, use_mask_tokens_attention, use_mask_self_attention, and set_timestep, and got the same result! 

I would appreciate it if you could help me clear up this confusion.

**Comment by AnsonLo11** [comment]
the code is based on diffusers==0.6.0 that is why no module named "CrossAttention"

**Comment by g-luo** [comment]
Yes, as stated by @AnsonLo11 we recommend that folks run this codebase with [diffusers==0.6.0](https://github.com/shape-guided-diffusion/shape-guided-diffusion/blob/b9fac837798b1c73110436ae763f07d164038620/environment.yaml#L13) to avoid issues with changes in the diffusers class structure.
